drivers/modem/modem_cellular.c

```diff
@@ -1861,3 +1950,61 @@ DT_INST_FOREACH_STATUS_OKAY(MODEM_CELLULAR_DEVICE_SWIR_HL7800)
 #define DT_DRV_COMPAT telit_me910g1
 DT_INST_FOREACH_STATUS_OKAY(MODEM_CELLULAR_DEVICE_TELIT_ME910G1)
 #undef DT_DRV_COMPAT
+
+int cellular_get_modem_info(const struct device *dev, enum cellular_modem_info_type type,
+			    char *info, size_t size)
+{
+	int rc = 0;
+
+	struct modem_cellular_data *data = (struct modem_cellular_data *)dev->data;
+
+	switch (type) {
+	case CELLULAR_MODEM_INFO_IMEI:
+		memcpy(info, &data->imei[0], MIN(size, sizeof(data->imei)));
+		break;
+	case CELLULAR_MODEM_INFO_SIM_IMSI:
+		memcpy(info, &data->imsi[0], MIN(size, sizeof(data->imsi)));
+		break;
+	case CELLULAR_MODEM_INFO_SIM_ICCID:
+		memcpy(info, &data->iccid[0], MIN(size, sizeof(data->iccid)));
+		break;
+	case CELLULAR_MODEM_INFO_MANUFACTURER:
+		memcpy(info, &data->manufacturer[0], MIN(size, sizeof(data->manufacturer)));
+		break;
+	case CELLULAR_MODEM_INFO_FW_VERSION:
+		memcpy(info, &data->fw_version[0], MIN(size, sizeof(data->fw_version)));
+		break;
+	case CELLULAR_MODEM_INFO_MODEL_ID:
+		memcpy(info, &data->model_id[0], MIN(size, sizeof(data->model_id)));
+		break;
+	default:
+		rc = -ENODATA;
+		break;
+	}
+
+	return rc;
+}
+
+int cellular_get_signal(const struct device *dev, const enum cellular_signal_type type,
+			int16_t *value)
+{
+	int rc = 0;
+
+	struct modem_cellular_data *data = (struct modem_cellular_data *)dev->data;
+
+	if ((data->state != MODEM_CELLULAR_STATE_AWAIT_REGISTERED) &&
+	    (data->state != MODEM_CELLULAR_STATE_CARRIER_ON)) {
+		return -ENODATA;
+	}
+	switch (type) {
+	case CELLULAR_SIGNAL_RSSI: {
+		*value = data->rssi;
```

Comment:
I think we should get this value on request instead of caching it, this way we will always return the latest value, and don't need to track if it is valid. We can run a script here sync when the function is called :)

Reply:
good idea, I'll make it generic so we could leverage this script to poll other network info